### ISSUE TITLE
jiva volume upgrade script sed fails

### ISSUE THREAD

**Issue by jp83** [openebs/openebs#2659]
## Description
I'm trying to upgrade from 0.9.0 to 1.0.0 following the instructions.  I cloned the repo and get to the ./jiva_volume_upgrade.sh step which fails due to sed option -u.  Seems like an obvious issue with the sed, but I can't find what the -u switch is supposed to do in the man.  I'm on MacOS, but didn't see anything different.

## Expected Behavior
Expected to upgrade jiva volume.

## Current Behavior
./jiva_volume_upgrade.sh pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032
pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032 is a jiva volume
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- u
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
Upgrading Replica Deployment to 1.0.0
error: must specify -p to patch
Failed to patch the deployment pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032-rep | Exit code: 1
Clearing temporary files

## Your Environment

helm list | grep openebs
openebs             	6       	Mon Jul  1 21:54:36 2019	DEPLOYED	openebs-1.0.0             	1.0.0                	openebs

kubectl get pv
pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032   5Gi        RWO            Delete           Bound    default/xxx       openebs-jiva-default            31h

**Comment by shubham14bajpai** [comment]
Thanks for the issue. The fix is merged #2654 . Please verify the changes.

**Comment by jp83** [comment]
The upgrade works now, but I think another similar issue with the verify script called within.

"pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032-rep-5758c8db5f" deleted
Upgrading target Deployment to 1.0.0
deployment.extensions/pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032-ctrl patched
replicaset.extensions "pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032-ctrl-569b6dd995" deleted
Upgrading target service to 1.0.0
service/pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032-ctrl-svc patched
Upgrade steps are done on volume pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032
verifying jiva pvc-6949cf81-9b6c-11e9-aaa1-000c297ba032 volume upgrade
Failed validation for controller pod not running
Failed validation for replica pods not running
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
                                                                       	[-e pattern] [-f file] [--binary-files=value] [--color=when]
                                                                                                                                    	[--context[=num]] [--directories=action] [--label] [--line-buffered]
                                            	[--null] [pattern] [file ...]
                                                                             usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]
                                                                                                                                                    	[-e pattern] [-f file] [--binary-files=value] [--color=when]
                                                    	[--context[=num]] [--directories=action] [--label] [--line-buffered]
                                                                                                                            	[--null] [pattern] [file ...]


it must be on a while loop because it keeps spitting out the above usage.

**Comment by wouterh** [comment]
the `-P` option for `grep` is not supported by the default `grep` on mac os x:
```
○ grep --version
grep (BSD grep) 2.5.1-FreeBSD
```

Replacing `-P` with `-E` works.

Also, it would be better if the scripts have:
```
#!/usr/bin/env bash
```

iso

```
#!/bin/bash
```

`/bin/bash` on mac os x is old:
```
○ /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)
Copyright (C) 2007 Free Software Foundation, Inc.
```
It doesn't support e.g. `declare -A`, which is used by `pre-upgrade.sh`. Changing the shebang allows to install a more recent bash (e.g. with homebrew) and have the scripts use it without modifications.

**Comment by wouterh** [comment]
On mac, I also run into the issue that `wc -l` prints spaces in front of the number:

```
running_ctrl_pod_count=$(kubectl get pods -n "$ns" \
-l openebs.io/controller=jiva-controller,openebs.io/persistent-volume="$pv" \
--no-headers | wc -l)
```
results in leading spaces in `running_ctrl_pod_count`:
```
○ echo "pods: $running_ctrl_pod_count"
pods:        1
```
which causes the comparison with 1 to fail.

**Comment by shubham14bajpai** [comment]
Thanks for reporting this. The fixes are available in master #2663  . Please verify the changes.
